src/bin/pgcopydb/ld_stream.h

```diff
@@ -275,6 +275,42 @@ typedef enum
 } LogicalStreamMode;
 
 
+/*
+ * Keep track of tables with generated columns to avoid unnecessary lookups
+ * in the catalog.
+ */
+typedef struct TableWithGeneratedColumns
+{
+	/* The table which has generated columns */
+	char nspname[PG_NAMEDATALEN];
+	char relname[PG_NAMEDATALEN];
+} TableWithGeneratedColumns;
+
+
+typedef struct GeneratedColumns
+{
+	char column[PG_NAMEDATALEN];
+
+	UT_hash_handle hh;           /* makes this structure hashable */
+} GeneratedColumns;
+
+
+/*
+ * This is a multi-level hash table, where the first level is the table which
+ * has generated columns and the second level is the column name itself.
+ */
+typedef struct GeneratedColumnsCache
+{
+	/* The key to lookup the generated columns */
+	TableWithGeneratedColumns table;
+
+	/* The generated columns cache */
+	GeneratedColumns *columns;
```

Comment:
Is this an array or a linked list or something else? I think the comments should say.